fabric-content-registries-v0/src/main/java/net/fabricmc/fabric/impl/content/registry/util/QuiltDeferringQueues.java

```diff
@@ -0,0 +1,181 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.content.registry.util;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.quiltmc.loader.api.QuiltLoader;
+import org.quiltmc.qsl.base.api.util.TriState;
+import org.quiltmc.qsl.block.content.registry.api.ReversibleBlockEntry;
+import org.quiltmc.qsl.registry.api.event.RegistryMonitor;
+import org.quiltmc.qsl.registry.attachment.api.RegistryEntryAttachment;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.item.Item;
+import net.minecraft.util.registry.Registry;
+import net.minecraft.world.event.GameEvent;
+
+public class QuiltDeferringQueues<T> {
+	private static final TriState CRASH_ON_DEFERRING_ENTRY = TriState.fromProperty("quilted_fabric_api.quilted_fabric_content_registries_v0.crash_on_deferring_entry");
+	private static final Logger LOGGER = LoggerFactory.getLogger("Quilted Fabric Content Registries");
+
+	public static final DeferringQueue<Block> BLOCK = new DeferringQueue<>(Registry.BLOCK);
+	public static final DeferringQueue<Item> ITEM = new DeferringQueue<>(Registry.ITEM);
+	public static final DeferringQueue<GameEvent> GAME_EVENT = new DeferringQueue<>(Registry.GAME_EVENT);
+
+	public static final Map<RegistryEntryAttachment<Object, Object>, List<RegistryEntryAttachment.Entry<Object, Object>>> OMNIQUEUE = new HashMap<>();
+
+	@SuppressWarnings("unchecked")
+	public static <K, V> void addEntry(RegistryEntryAttachment<K, V> queue, K entry, V value) {
+		boolean hasDeferredKey = deferEntry(entry);
+		boolean hasDeferredValue = deferEntry(value);
+
+		if (hasDeferredKey || hasDeferredValue) {
+			if (!OMNIQUEUE.containsKey(queue)) {
+				OMNIQUEUE.put((RegistryEntryAttachment<Object, Object>) queue, new ArrayList<>());
+			}
+
+			OMNIQUEUE.get(queue).add(new RegistryEntryAttachment.Entry<>(entry, value));
+		} else {
+			queue.put(entry, value);
+		}
+	}
+
+	public static <V> boolean deferEntry(V value) {
+		if (value instanceof Block block) {
+			QuiltDeferringQueues.BLOCK.deferEntry(block);
+		} else if (value instanceof BlockState state) {
+			QuiltDeferringQueues.BLOCK.deferEntry(state.getBlock());
+		} else if (value instanceof ReversibleBlockEntry entry) {
+			return QuiltDeferringQueues.BLOCK.deferEntry(entry.block());
+		} else if (value instanceof Item item) {
+			QuiltDeferringQueues.ITEM.deferEntry(item);
+		} else if (value instanceof GameEvent event) {
+			QuiltDeferringQueues.GAME_EVENT.deferEntry(event);
+		} else {
+			return false;
+		}
+
+		return true;
```

Comment:
...okay, the flow of this method confuses me.
```suggestion
		if (value instanceof Block block) {
			QuiltDeferringQueues.BLOCK.deferEntry(block);
			return true;
		} else if (value instanceof BlockState state) {
			QuiltDeferringQueues.BLOCK.deferEntry(state.getBlock());
			return true;
		} else if (value instanceof ReversibleBlockEntry entry) {
			return QuiltDeferringQueues.BLOCK.deferEntry(entry.block());
		} else if (value instanceof Item item) {
			QuiltDeferringQueues.ITEM.deferEntry(item);
			return true;
		} else if (value instanceof GameEvent event) {
			QuiltDeferringQueues.GAME_EVENT.deferEntry(event);
			return true;
		}

		return false;
```
This might not be DRY, but... c'mon, it's a `return primitive` statement.